### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Shevtsooov**
[DEMO LINK](https://Shevtsooov.github.io/react_list-of-goods/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -26,70 +27,121 @@ type ReorderOptions = {
...
-  return visibleGoods;
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed ? visibleGoods.reverse() : visibleGoods;
```
Comment (naziamoff) [inline src/App.tsx:49]: ternaries should be multilined

File: src/App.tsx [hunk 2]
```diff
@@ -26,70 +27,121 @@ type ReorderOptions = {
...
+          <button
+            type="button"
+            className={cn(
+              'section__button',
+              'section__button--alphabet',
+              'is-info',
+              { 'is-light generalButton': sortType !== SortType.ALPHABET },
+            )}
+            onClick={() => {
+              this.setState({ sortType: SortType.ALPHABET });
```
Comment (naziamoff) [inline src/App.tsx:82]: you should create a method for that

File: src/App.tsx [hunk 3]
```diff
@@ -26,70 +27,121 @@ type ReorderOptions = {
...
+            type="button"
+            className={cn(
+              'section__button',
+              'section__button--length',
+              'is-success',
+              { 'is-light generalButton': sortType !== SortType.LENGTH },
+            )}
+            onClick={() => {
+              this.setState({ sortType: SortType.LENGTH });
+            }}
```
Comment (naziamoff) [inline src/App.tsx:98]: separate method

File: src/App.tsx [hunk 4]
```diff
@@ -26,70 +27,121 @@ type ReorderOptions = {
...
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={cn(
+              'section__button',
+              'section__button--reverse',
+              'is-warning',
+              { 'is-light generalButton': !isReversed },
```
Comment (naziamoff) [inline src/App.tsx:109]: cn formatting

File: src/App.tsx [hunk 5]
```diff
@@ -26,70 +27,121 @@ type ReorderOptions = {
...
+            )}
+            onClick={() => {
+              this.setState({ isReversed: !isReversed });
+            }}
```
Comment (naziamoff) [inline src/App.tsx:113]: separate method

File: src/App.tsx [hunk 6]
```diff
@@ -26,70 +27,121 @@ type ReorderOptions = {
...
+          >
+            Reverse
+          </button>
+
+          {(sortType !== SortType.NONE || isReversed) && (
```
Comment (naziamoff) [inline src/App.tsx:118]: create a separate variable for that

File: src/App.tsx [hunk 7]
```diff
@@ -26,70 +27,121 @@ type ReorderOptions = {
...
+            <button
+              type="button"
+              className="
+              section__button
+              section__button--reset
+              is-danger
+              is-light
+              "
```
Comment (naziamoff) [inline src/App.tsx:126]: not bad, but don't forget about indentation of the child elements
```suggestion
              className="
                section__button
                section__button--reset
                is-danger
                is-light
              "
```

File: src/App.tsx [hunk 8]
```diff
@@ -26,70 +27,121 @@ type ReorderOptions = {
...
+              onClick={() => {
+                this.setState({
+                  isReversed: false,
+                  sortType: SortType.NONE,
+                });
```
Comment (naziamoff) [inline src/App.tsx:131]: separate method

**Review by naziamoff — CHANGES_REQUESTED** [review]

**Comment by Shevtsooov** [comment]
@naziamoff Hi, thanks for the review!
All the issues have been fixes, please check.

**Review by naziamoff — APPROVED** [review]
awesome job 🔥

**Pull request closed without merging**